src/ServiceProviders/RepositoryProviders/Bitbucket/BitbucketProvider.php

```diff
@@ -168,4 +170,55 @@ public function getSecretValues() {
       ];
     }
 
+    protected function getProjectOptions($org)
+    {
+        // Trailing slash is REQUIRED on this resource. See https://jira.atlassian.com/browse/BCLOUD-17211
+        $projects = $this->api()->pagedRequest("teams/$org/projects/");
+        if (empty($projects)) {
+            return [];
+        }
+        $options = [];
+        foreach ($projects as $value) {
+            $options[] = [
+                "name" => $value['name'],
+                "key" => $value['key'],
+                "uuid" => $value['uuid'],
+            ];
+        }
+        return $options;
+    }
+
+    /**
+     * Add repository to a project, if bitbucket org supports it.
+     */
+    public function addInteractions(InputInterface $input, OutputInterface $output)
+    {
+        if (!$org = $input->getOption('org')) {
+            return;
+        }
+        $extras = $input->getOption('extra');
+        if (!empty($extras['bitbucket-project'])) {
+            $this->project = $extras['bitbucket-project'];
```

Comment:
For this bit, i was thinking we could search project_options for key, and uuid (maybe name as well, but only if it's unique - not sure offhand)
Rather than assume / force user to specify UUID, this would allow callers to specify any of those options.

Reply:
I had that same thought. I have a few meetings but will take a pass at that when I have some more flex time